CONTRIBUTING.md

```diff
@@ -70,7 +74,69 @@ will be posted in the issue's thread.
 * Bulleted points are fine.
 * Typically a hyphen or asterisk is used for the bullet, followed by a single space.
 
-### Merging Commits
+## Pull Request Process
+
+### Creating a Pull Request
+
+Use the pull request template to provide a complete description of the change.
+The template aims to capture important information to streamline the review
+process, ensure your changes are captured in release notes, and update related
+issues. Your pull request description and any discussion that follows is a
+contribution itself that will help the community and future contributors
+understand the project better.
+
+* Before submitting a pull request, please make sure you verify the changes
+  locally. The `Makefile` in this repository provides useful targets such as
+  `lint` and `test` to make verification easier.
+* Prefer small commits and small pull requests over large changes.
+  Small changes are easier to digest and get reviewed faster. If you find
+  that your change is getting large, break up your PR into small, logical
+  commits. Consider breaking up large PRs into smaller PRs, if each of them
+  can be useful on its own.
+* Have good commit messages. Please see [Commit Messages](#commit-messages)
+  section for guidelines.
+* Pull requests *should* reference an existing issue and include a `Fixes #NNNN`
+  or `Updates #NNNN` comment. Remember that `Fixes` will close the associated
+  issue, and `Updates` will link the PR to it.
+
+### Getting your Pull Request Reviewed, Approved, and Merged
+
+Once a pull request has been opened, the following must take place before it is merged:
+
+* It needs the `ok-to-merge` label
+* It needs to pass all the checks.
+* It needs to be approved by a [CODEOWNER](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CODEOWNERS) for all files changed
+
+While these steps will not always take place in the same order, the following describes the process for a typical pull request once it is opened:
+
+1. Review is automatically requested from CODEOWNERS and `needs-triage` label is automatically added.
+2. Assignee is added to pull request to ensure it gets proper attention throughout the process.
+   Typically one of the CODEOWNERS will assign themselves, but they may choose to delegate to someone else.
+3. Triage removes `needs-triage` and adds `ok-to-merge` if the pull request is generally aligned with product goals and does not conflict with current milestones; otherwise they may add a comment and a `do-not-merge/*` label.
```

Comment:
I had to read this line a couple of times to make sure I knew who `Triage` was. I _assume_, based on discussion in SIG meetings and context clues, that `Triage` is referring to a group of tanzu-framework product owners. Do you think there is any chance we need to be more specific about who `Triage` is so that community members know exactly who is on the hook for this step?

Reply:
@ankeesler I think it could be some combination of: 1) the assignee, 2) tanzu-framework-reviewers, or 3) a person wearing the "triage" hat during some periodic triage meeting... without more context on how that is evolving, I was weary of being overly specific here.

In the meantime, my hope is that this final paragraph offers enough clarity on what to do if things are not progressing:

> If any of the above is unclear, and there has been no new activity for 2-3 days,
the contributor is encouraged to seek further information by commenting and
mentioning the assignee or @vmware-tanzu/tanzu-framework-reviewers if there is
no assignee or they themselves are unresponsive.

cc @vuil

Reply:
Yeah we can iterate and provider more clarifications when the triage process is more nailed down. For now, I think the quoted paragraph is a useful catch-all on what actions to take when things are not progressing.